lib/config/formatter.go

```diff
@@ -40,65 +40,64 @@ type textFormatter struct {
 	// FormatCaller is a function to return (part) of source file path for output.
 	// Defaults to filePathAndLine() if unspecified
 	FormatCaller func() (caller string)
-	// InitFormat is the initial format of log configuration
-	InitFormat []string
 	// InputFormat is the parsed format of each log line
-	InputFormat []string
+	LogFormat []string
+	// TimestampEnabled specifies if timestamp is enabled in logs
+	TimestampEnabled bool
+	// CallerEnabled specifies if caller is enabled in logs
+	CallerEnabled bool
 }
 
-// DefaultLoggerFormat is the default format in which to display logs (i.e. log.format in teleport.yaml is empty)
-var DefaultLoggerFormat = []string{"level", "component", "message", "caller"}
-var re = regexp.MustCompile(`[a-z]*`)
-
 type writer struct {
 	bytes.Buffer
 }
 
 const (
-	noColor = -1
-	red     = 31
-	yellow  = 33
-	blue    = 36
-	gray    = 37
+	noColor        = -1
+	red            = 31
+	yellow         = 33
+	blue           = 36
+	gray           = 37
+	levelField     = "level"
+	componentField = "component"
+	callerField    = "caller"
+	timestampField = "timestamp"
 )
 
+// TextFormatterFields are the fields for log format
+var TextFormatterFields = []string{"level", "component", "caller", "timestamp"}
+
 // CheckAndSetDefaults checks and sets log format configuration
 func (tf *textFormatter) CheckAndSetDefaults() error {
 	// set padding
 	if tf.ComponentPadding == 0 {
 		tf.ComponentPadding = trace.DefaultComponentPadding
 	}
 	// set caller
-	if tf.FormatCaller == nil {
-		tf.FormatCaller = formatCallerWithPathAndLine
-	}
+	tf.FormatCaller = formatCallerWithPathAndLine
+
 	// set log formatting
-	if tf.InitFormat == nil {
-		tf.InputFormat = DefaultLoggerFormat
+	if tf.LogFormat == nil {
+		tf.TimestampEnabled = true
+		tf.CallerEnabled = true
+		tf.LogFormat = KnownFormatFields.names()
 		return nil
 	}
 	// parse input
-	res, err := parseInputFormat(tf.InitFormat)
+	res, err := parseInputFormat(tf.LogFormat)
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	tf.InputFormat = res
-	return nil
-}
+	if contains(res, timestampField) {
+		tf.TimestampEnabled = true
```

Comment:
I'd also keep the formatter's internal attributes unexported unless this becomes really necessary.